bluepill/src/BPPacker.m

```diff
@@ -97,6 +97,42 @@ @implementation BPPacker
     return bundles;
 }
 
++ (NSMutableArray<BPXCTestFile *> *)splitXCTestBundles:(NSArray<BPXCTestFile *> *)xcTestFiles
+                                   withTestsByFilePath:(NSMutableDictionary *)testsToRunByFilePath
+                           withTestEstimatesByFilePath:(NSMutableDictionary *)testEstimatesByFilePath
+                                         withTestTimes:(NSDictionary *)testTimes
+                                 withOptimalBundleTime:(double)optimalBundleTime {
+    NSMutableArray<BPXCTestFile *> *bundles = [[NSMutableArray alloc] init];
+    for (BPXCTestFile *xctFile in xcTestFiles) {
+        NSArray *bundleTestsToRun = [[testsToRunByFilePath[xctFile.testBundlePath] allObjects] sortedArrayUsingSelector:@selector(compare:)];
+        NSNumber *estimatedBundleTime = (NSNumber *)testEstimatesByFilePath[xctFile.testBundlePath];
+        if ([estimatedBundleTime integerValue] < optimalBundleTime) {
+            BPXCTestFile *bundle = [self makeBundle:xctFile withTests:bundleTestsToRun startAt:0 numTests:[bundleTestsToRun count] estimatedTime:estimatedBundleTime];
+            [bundles addObject:bundle];
+            continue;
+        }
+        double sumOfSplits = 0.0;
+        for (int i = 0; i < [bundleTestsToRun count];) {
+            double splitExecTime = 0.0;
+            NSUInteger startIndex = i;
+            while(splitExecTime < optimalBundleTime && i < [bundleTestsToRun count]) {
+                NSString *test = [bundleTestsToRun objectAtIndex:i];
+                if ([testTimes objectForKey:test]) {
+                    splitExecTime += [testTimes[test] doubleValue];
+                }
+                i++;
+            }
+            // Make a bundle out of current xctFile
+            BPXCTestFile *bundle = [self makeBundle:xctFile withTests:bundleTestsToRun startAt:startIndex numTests:(i-startIndex) estimatedTime:[NSNumber numberWithDouble:splitExecTime]];
+            [bundles addObject:bundle];
+            sumOfSplits += splitExecTime;
```

Comment:
Is sumOfSplits being used elsewhere?

Reply:
Its log statement is gone, so I cleaned it up.